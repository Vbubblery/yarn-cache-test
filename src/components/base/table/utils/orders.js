import { frozenPeriods, isFrozen, isMoq, isLotSize } from "./planning";
import { defaultTableValueFormatting, isNotPast } from "./base.js";

export const ORDER_STATUS = {
  VALIDATED: "validated",
  LOCKED: "locked"
};
export const ORDER_FREEZE_STATUS = {
  PARTIAL: "partlyFrozen",
  TOTAL: "fullyFrozen"
};

export const getOrderStatus = status => {
  switch (status) {
    case 2:
      return ORDER_STATUS.VALIDATED;
    case 3:
    case 4:
      return ORDER_STATUS.LOCKED;
    default:
      return undefined;
  }
};

export const getOptionsForOrders = options => {
  const { leadTimes, periods, periodType, productData } = options;
  return {
    periodType,
    periods,
    frozenPeriods: frozenPeriods(
      productData?.leadTime || {},
      periods,
      periodType,
      leadTimes
    ),
    isAggregated: options.isAggregated,
    access: options.access,
    leadTimes: options.leadTimes,
    units: options.productData?.units,
    moq: options.productData?.moq,
    lotSize: options.productData?.lotSize
  };
};

export const getOptionsForStatus3Supplies = options => {
  const { periods, periodType, isAggregated, access } = options;
  return {
    periodType,
    periods,
    isAggregated,
    access,
    units: options.productData?.units
  };
};

export const getOptionsForDemand = options => ({
  periodType: options.periodType,
  periods: options.periods,
  isAggregated: options.isAggregated,
  access: options.access
});

export const getOrdersValues = (orders, options) => {
  if (orders.isParent) {
    return orders.values.map((value, index) => ({
      value,
      isPast: index === 0
    }));
  }

  const {
    isAggregated,
    moq: moqBySupplier,
    lotSize: lotSizeBySupplier,
    ...otherOptions
  } = options;
  const { editedBy, orderStatus, supplier, isParent } = orders;
  return orders.values.map((value, index) => {
    const status = orderStatus && getOrderStatus(orderStatus[index]);
    const isByFlowlity = editedBy && editedBy[index] === "flowlity";
    const frozenStatus = isFrozen(index, supplier, options);
    const moq = isMoq(supplier, isAggregated, orders, moqBySupplier);
    const lotSize = isLotSize(
      supplier,
      isAggregated,
      orders,
      lotSizeBySupplier
    );
    return {
      value,
      isByFlowlity,
      status,
      moq,
      lotSize,
      isPast: index === 0,
      tooltip: supplier !== null || supplier !== undefined,
      isEditable: options.access && !isAggregated && !isParent && index > 0,
      ...(!isNaN(value) && { isFrozen: frozenStatus }),
      ...otherOptions
    };
  });
};

export const getStatus3SuppliesValues = (supplies, options) => {
  const { isAggregated, access, ...otherOptions } = options;
  return supplies.values.map((value, index) => {
    return {
      value,
      isPast: index === 0,
      isEditable: access && !isAggregated && index > 0,
      ...otherOptions
    };
  });
};

export const getAllOrdersValues = (allOrders, options) => {
  return allOrders.values.map((value, index) => {
    return {
      value,
      isPast: !isNotPast(index, options.periods, options.periodType)
    };
  });
};

export const allOrdersCalc = (value, isPast) => {
  if (isPast) return null;
  else if (!value) return 0;
  return defaultTableValueFormatting(value);
};
