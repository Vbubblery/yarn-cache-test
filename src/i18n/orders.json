{
  "en": {
    "orderModalAddProductsButton": "Add products",
    "constraintsNotMetMessage": "Some of the products don't meet their constraints",
    "moq": "Minimum order quantity",
    "constraint": "Constraint | Constraint | Constraints",
    "orderConstraints": "Order constraints",
    "orderModalTitle": "Order for",
    "editingModalTitle": "Editing order for",
    "orderTitle": "Orders",
    "planned": "Purchase requisition | Purchase requisition | Purchase requisitions",
    "validated": "Confirmed purchase requisition | Confirmed purchase requisition | Confirmed purchase requisitions",
    "purchaseOrders": "Purchase order | Purchase order | Purchase orders",
    "salesOrders": "Sales order | Sales order | Sales orders",
    "delivered": "Delivered | Delivered | Delivered",
    "search": "Search by order ID, supplier or product",
    "ship_before": "Ship before",
    "send_before": "Send before",
    "date": "Delivery date",
    "totalQuantity": "Total quantity",
    "quantity": "Quantity",
    "addedItems": "Added items",
    "groupedBy": "Grouped by",
    "inTotal": "In total",
    "download_selected": "Download",
    "upload_selected": "Send to API",
    "sftp_selected": "Export to SFTP",
    "sftp_exported": "Exported to SFTP",
    "temporary_id": "Temporary ID",
    "order_id": "Order ID",
    "select_all": "Select all",
    "deselect_all": "Deselect all",
    "deleteSelected": "Delete selected",
    "noOrders": "No orders found",
    "delivery": "Delivery",
    "constraintFrequency": "Doesn't meet order frequency requirements",
    "constraintMOQ": "Doesn't meet MOQ requirements",
    "constraintFullTruck": "Doesn't meet full truck requirements",
    "issueMOQ": "product have issue with MOQ or lot size | products have issue with MOQ or lot size",
    "orderIssues": "Order issues",
    "productIssues": "Product issues",
    "days": "days",
    "latestOrderDate": "Latest Order Date",
    "deliveryDate": "Delivery Date",
    "deselectAll": "Deselect all",
    "suppliers": "Suppliers",
    "reset": "Reset",
    "dateRange": "Date range",
    "thisWeek": "This week",
    "in2Weeks": "In 2 weeks",
    "in1month": "In 1 month",
    "in2months": "In 2 months",
    "in3months": "In 3 months",
    "tempId": "Temporary ID",
    "orderTitleDetail": "Details",
    "orderTitleStatus1": "Purchase requisition",
    "orderTitleStatus2": "Confirmed purchase requisition",
    "orderTitleStatus3": "Purchase order",
    "orderTitleStatus4": "Delivered order",
    "orderDetailMessageTitle": "Messages",
    "orderId": "Order ID",
    "day": "day | day | days",
    "value": "Value",
    "product": "Product",
    "reference": "Reference",
    "lotSize": "Lot Size",
    "leadTime": "Delivery lead time",
    "daysOfCoverage": "Stock coverage at delivery",
    "goodMoq": "Min. order quantity",
    "orderFrequency": "Order frequency",
    "badFullTruckQuantity": "The order is not respecting the full truck quantity constraint",
    "badOrderFrequency": "The order is not respecting the frequency constraint",
    "fullTruckQty": "Full truck quantity",
    "badMOQSupply": "The supply is not respecting the MOQ constraint",
    "badMOQOrder": "The order is not respecting the MOQ constraint",
    "expectedDelivery": "Expected delivery",
    "delete": "Delete",
    "cancel": "Cancel",
    "edit": "edit",
    "cancelEdit": "cancel edit",
    "discardChanges": "Discard Changes",
    "save": "Save",
    "orderIsUpdating": "Order is updating...",
    "deleteSuccess": "Order has been deleted",
    "deleteOrderError": "Cannot delete the order",
    "deleteSupplyError": "The product could not be deleted correctly",
    "deleteWarningTitle": "Delete confirmation",
    "deleteCurrentOrderWarningText": "Are you sure you want to remove this order?",
    "deleteCurrentOrderSupplyWarningText": "This will delete all the planned supplies inside this order.",
    "deleteSupplyWarningText": "Are you sure you want to remove this product ?",
    "deleteWarningConfirmButton": "YES, CONFIRM",
    "deleteWarningCancelButton": "CANCEL",
    "mergeWarningTitle": "Merge confirmation",
    "mergeWarningText": "An order already exists for this date and supplier. Do you want to combine the two orders?\n This will add the current products and quantities to the order found at this new date.",
    "mergeWarningConfirmButton": "YES, CONFIRM",
    "mergeWarningCancelButton": "CANCEL",
    "secondaryUnit": "Secondary unit",
    "orderAndSupplyConstraintNotMet": "Some constraints on the order level and product level are not met",
    "orderConstraintNotMet": "Some constraints on the order level are not met",
    "supplyConstraintNotMet": "Some products don't meet their constraints",
    "customer": {
      "display": "Customer",
      "partner": "Supplier",
      "waitValidateOrder": "Waiting to be validated",
      "orderValidated": "Validated",
      "waitSendOrder": "Waiting to be sent",
      "orderSent": "Sent",
      "sendBefore": "Send before",
      "badLotSize": "Lot size not respected",
      "validateBtn": "Validate"
    },
    "supplier": {
      "display": "Supplier",
      "partner": "Customer",
      "waitValidateOrder": "Waiting to be accepted",
      "orderValidated": "Accepted",
      "waitSendOrder": "Waiting to be shipped",
      "orderSent": "Shipped",
      "leadTimeNotSet": "Delivery lead time is not defined. Set it.",
      "sendBefore": "Ship before",
      "badLotSize": "Lot size not respected",
      "validateBtn": "Accept"
    }
  },
  "fr": {
    "orderModalTitle": "Commande pour",
    "editingModalTitle": "Édition de la commande pour",
    "constraintsNotMetMessage": "Certains produits ne respectent pas leurs contraintes",
    "moq": "Minimum de commande",
    "constraint": "Contrainte | Contrainte | Contraintes",
    "orderConstraints": "Contraintes de commande",
    "orderModalAddProductsButton": "Ajouter des produits",
    "orderTitle": "Commandes",
    "planned": "Prévue | Prévue | Prévues",
    "validated": "Prévalidée | Prévalidée | Prévalidées",
    "purchaseOrders": "Commande d'achat | Commande d'achat | Commandes d'achat",
    "salesOrders": "Commande de vente | Commande de vente | Commandes de vente",
    "delivered": "Livrée | Livrée | Livrées",
    "search": "Chercher par ID, fournisseur ou produit",
    "ship_before": "Expédier le",
    "send_before": "Passer commande avant le",
    "date": "Date de livraison prévue",
    "quantity": "Quantité",
    "totalQuantity": "Quantité totale",
    "addedItems": "Produits ajoutés",
    "groupedBy": "Groupé par",
    "inTotal": "Total",
    "download_selected": "Télécharger",
    "upload_selected": "Envoyer à l'API",
    "sftp_selected": "Exporter VIA SFTP",
    "sftp_exported": "Exportée via SFTP",
    "temporary_id": "ID provisoire",
    "order_id": "ID de la commande",
    "select_all": "Tout sélectionner",
    "deselect_all": "Tout désélectionner",
    "deleteSelected": "Supprimer la sélection",
    "noOrders": "Pas de commande trouvée",
    "delivery": "Livraison le",
    "constraintFrequency": "Ne respecte pas la contrainte de fréquence de commande",
    "constraintMOQ": "Ne respecte pas la contrainte de MOQ",
    "constraintFullTruck": "Ne respecte pas la contrainte de camion plein",
    "issueMOQ": " produit a un problème de MOQ ou de taille de lot | produits ont un problème de MOQ ou de taille de lot",
    "orderIssues": "Lié à la commande",
    "productIssues": "Lié aux produits",
    "days": "jours",
    "latestOrderDate": "Date de commande",
    "deliveryDate": "Date de livraison",
    "deselectAll": "Tout désélectionner",
    "suppliers": "Fournisseurs",
    "reset": "Réinitialiser",
    "dateRange": "Période",
    "thisWeek": "Cette semaine",
    "in2Weeks": "Dans 2 semaines",
    "in1month": "Dans 1 mois",
    "in2months": "Dans 2 mois",
    "in3months": "Dans 3 mois",
    "noTimeRestriction": "Pas de limitation",
    "lastOrderDate": "Date de la commande",
    "tempId": "ID provisoire",
    "orderId": "ID commande",
    "orderTitleDetail": "Détails",
    "orderTitleStatus1": "Commande prévue",
    "orderTitleStatus2": "Commande prévalidée",
    "orderTitleStatus3": "Commande d'achat",
    "orderTitleStatus4": "Commande livrée",
    "orderDetailMessageTitle": "Messages",
    "day": "jour | jour | jours",
    "value": "Valeur",
    "product": "Produit",
    "reference": "Référence",
    "lotSize": "Taille de lot",
    "leadTime": "Délai de livraion",
    "daysOfCoverage": "Couverture de stock à réception",
    "goodMoq": "Minimum de commande",
    "orderFrequency": "Fréquence de commande",
    "badFullTruckQuantity": "La contrainte de camion plein n'est pas respectée",
    "badOrderFrequency": "La contrainte de fréquence de commande n'est pas respectée",
    "fullTruckQty": "Camion plein",
    "badMOQSupply": "La quantité minimum n'est pas respectée",
    "badMOQOrder": "Le minimum de commande n'est pas respecté",
    "expectedDelivery": "Livraison attendue",
    "delete": "Supprimer",
    "cancel": "Annuler",
    "edit": "éditer",
    "cancelEdit": "annuler l'édition",
    "discardChanges": "Annuler les modifications",
    "save": "Enregistrer",
    "orderIsUpdating": "La commande est mise à jour...",
    "deleteSuccess": "Commande supprimée",
    "deleteOrderError": "Suppression impossible",
    "deleteSupplyError": "Une erreur est survenue lors de la suppression du produit",
    "deleteWarningTitle": "Confirmation de suppression",
    "deleteCurrentOrderWarningText": "Voulez-vous supprimer cette commande ?",
    "deleteCurrentOrderSupplyWarningText": "Cela va supprimer tous les approvisionnements liés à cette commande",
    "deleteSupplyWarningText": "Etes vous sûr de vouloir supprimer ce produit ?",
    "deleteWarningConfirmButton": "JE CONFIRME",
    "deleteWarningCancelButton": "ANNULER",
    "mergeWarningTitle": "Confirmation de combinaison",
    "mergeWarningText": "Une commande existe déjà pour cette date et fournisseur. Voulez-vous combiner les deux commandes?\n Ceci aura pour effet d'ajouter les produits et quantités à la commande trouvée à cette nouvelle date.",
    "mergeWarningConfirmButton": "JE CONFIRME",
    "mergeWarningCancelButton": "ANNULER",
    "secondaryUnit": "Unité secondaire",
    "orderAndSupplyConstraintNotMet": "Certaines contraintes au niveau commande et au niveau produit ne sont pas respectées",
    "orderConstraintNotMet": "Certaines contraintes au niveau commande ne sont pas respectées",
    "supplyConstraintNotMet": "Certains produits ne respectent pas leurs contraintes",
    "customer": {
      "display": "Client",
      "partner": "Fournisseur",
      "waitValidateOrder": "En attente de validation",
      "orderValidated": "Validé",
      "waitSendOrder": "En attente d'expédition",
      "sendBefore": "Expédier avant",
      "orderSent": "Expédié",
      "badLotSize": "La taille du lot n'est pas respectée",
      "validateBtn": "Valider"
    },
    "supplier": {
      "display": "Fournisseur",
      "partner": "Client",
      "waitValidateOrder": "En attente de validation",
      "orderValidated": "Accepté",
      "waitSendOrder": "En attente d'expédition",
      "leadTimeNotSet": "Le délai de livraison n'est pas défini. Configurer.",
      "sendBefore": "Expédier avant",
      "orderSent": "Expédié",
      "badLotSize": "La taille du lot n'est pas respectée",
      "validateBtn": "Accepter"
    }
  }
}
