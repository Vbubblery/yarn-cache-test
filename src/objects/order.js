import { validate } from "jsonschema";
import OrderSchema, { orderSupplySchema } from "@/schemas/order";

export class OrderSupply {
  constructor(model = {}) {
    this._model = model;
    this._id = model.id;
    this._productId = model.product_site;
    this._externalId = model.external_id;
    this._isDefaultUnit = model.is_default_unit;
    this._defaultUnitValue = model.default_unit_value;
    this._quantity = model.quantity;
    this._secondaryUnitValue = model.second_unit_value;
    this._secondaryUnitQuantity = model.second_unit_quantity;
    this._orderId = model.order;
    this._name = model.name;
    this._unit = model.unit;
    this._secondaryUnit = model.second_unit;
    this._leadTime = model.lead_time;
    this._price = model.product_price;
    this._totalPrice = model.total_price;
    this._moq = model.moq;
    this._description = model.description;
    this._lotSize = model.lot_size;
    this._stockAtEndOfPeriod = model.stock_at_end_of_period;
    this._minStockAtEndOfPeriod = model.minimum_stock_at_end_of_period;
    this._totalDemand = model.total_demand;
  }
  static create(model) {
    return validate(model, OrderSupply.schema())
      ? new OrderSupply(model)
      : null;
  }

  static schema() {
    return orderSupplySchema;
  }

  get externalId() {
    return this._externalId;
  }

  get defaultUnitValue() {
    return this._defaultUnitValue;
  }

  get secondaryUnitValue() {
    return this._secondaryUnitValue;
  }

  get secondaryUnitQuantity() {
    return this._secondaryUnitQuantity;
  }

  get secondaryUnit() {
    return this._secondaryUnit;
  }

  get moq() {
    return this._moq;
  }

  get description() {
    return this._description;
  }

  get lotSize() {
    return this._lotSize;
  }

  get id() {
    return this._id;
  }

  get productId() {
    return this._productId;
  }

  get quantity() {
    return this._quantity;
  }

  get orderId() {
    return this._orderId;
  }

  get name() {
    return this._name;
  }

  get unit() {
    return this._unit;
  }

  get leadTime() {
    return this._leadTime;
  }

  get price() {
    return this._price;
  }

  get totalPrice() {
    return this._totalPrice;
  }

  get isDefaultUnit() {
    return this._isDefaultUnit;
  }

  get minStockAtEndOfPeriod() {
    return this._minStockAtEndOfPeriod;
  }
  get stockAtEndOfPeriod() {
    return this._stockAtEndOfPeriod;
  }
  get totalDemand() {
    return this._totalDemand;
  }
  set id(id) {
    this._id = id;
  }

  set productId(productId) {
    this._productId = productId;
  }

  set orderId(orderId) {
    this._orderId = orderId;
  }

  set quantity(quantity) {
    this._quantity = quantity;
  }

  set name(name) {
    this._name = name;
  }

  set unit(unit) {
    this._unit = unit;
  }

  set leadTime(leadTime) {
    this._leadTime = leadTime;
  }
  set totalPrice(totalPrice) {
    this._totalPrice = totalPrice;
  }

  add(otherSupply) {
    return new OrderSupply({
      ...this._model,
      quantity: this.quantity + otherSupply.quantity
    });
  }
}

export class Order {
  constructor(model = {}) {
    this._customerSiteId = model.customer_site_id;
    this._customerSiteName = model.customer_site_name;
    this._date = model.date;
    this._externalId = model.external_id;
    this._fullTruckQuantity = model.full_truck_quantity;
    this._fullTruckUnit = model.full_truck_unit;
    this._id = model.id;
    this._minOrderQuantity = model.minimum_order_quantity;
    this._minOrderUnit = model.minimum_order_unit;
    this._orderFrequency = model.order_frequency;
    this._previousOrderDate = model.previous_order_date;
    this._sendBefore = model.send_before;
    this._shipBefore = model.ship_before;
    this._statusId = model.status_id;
    this._supplierId = model.supplier_id;
    this._supplies = (model.supplies || []).map(supply =>
      OrderSupply.create(supply)
    );
    this._supplierName = model.name;
    this._tagGrouper = model.tag_grouper;
    this._leadTime = model.lead_time;
    this._lastSftpExportDate = model.last_sftp_export_date;
  }
  static create(model) {
    return validate(model, Order.schema()) ? new Order(model) : null;
  }
  static schema() {
    return OrderSchema;
  }

  // Getters
  get customerSiteId() {
    return this._customerSiteId;
  }
  get customerSiteName() {
    return this._customerSiteName;
  }
  get orderFrequency() {
    return this._orderFrequency;
  }
  get previousOrderDate() {
    return this._previousOrderDate;
  }
  get sendBefore() {
    return this._sendBefore;
  }
  get shipBefore() {
    return this._shipBefore;
  }
  get id() {
    return this._id;
  }
  get statusId() {
    return this._statusId;
  }
  get date() {
    return this._date;
  }
  get supplierId() {
    return this._supplierId;
  }
  get supplies() {
    return this._supplies;
  }
  get fullTruckQuantity() {
    return this._fullTruckQuantity;
  }
  get fullTruckUnit() {
    return this._fullTruckUnit;
  }
  get minOrderQuantity() {
    return this._minOrderQuantity;
  }
  get minOrderUnit() {
    return this._minOrderUnit;
  }
  get supplierName() {
    return this._supplierName;
  }
  get externalId() {
    return this._externalId;
  }
  get tagGrouper() {
    return this._tagGrouper;
  }
  get leadTime() {
    return this._leadTime;
  }
  get lastSftpExportDate() {
    return this._lastSftpExportDate;
  }

  // Setters
  set id(id) {
    this._id = id;
  }

  set statusId(statusId) {
    this._statusId = statusId;
  }
  set date(date) {
    this._date = date;
  }
  set supplierId(supplierId) {
    throw new Error(
      "Cannot change a supplier for an existing order. Create a new order instead"
    );
  }
  set supplies(supplies) {
    this._supplies = supplies;
  }
  set fullTruckQuantity(fullTruckQuantity) {
    this._fullTruckQuantity = fullTruckQuantity;
  }
  set fullTruckUnit(fullTruckUnit) {
    this._fullTruckUnit = fullTruckUnit;
  }
  set minOrderQuantity(minOrderQuantity) {
    this._minOrderQuantity = minOrderQuantity;
  }
  set minOrderUnit(minOrderUnit) {
    this._minOrderUnit = minOrderUnit;
  }
  set supplierName(supplierName) {
    this._supplierName = supplierName;
  }
  set externalId(externalId) {
    this._externalId = externalId;
  }
  set tagGrouper(tagGrouper) {
    this._tagGrouper = tagGrouper;
  }
  set lastSftpExportDate(lastSftpExportDate) {
    this._lastSftpExportDate = lastSftpExportDate;
  }
}
