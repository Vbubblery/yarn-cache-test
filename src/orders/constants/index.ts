export enum ORDER_TYPE {
  NEW_ORDERS = "new_orders",
  PLANNED = "planned",
  VALIDATED = "validated",
  PURCHASE_ORDER = "purchaseOrders",
  SALES_ORDER = "salesOrders",
  DELIVERED = "delivered"
}

export enum ORDER_TYPE_VALUE {
  PLANNED = 1,
  VALIDATED = 2,
  PURCHASE_ORDER = 3,
  SALES_ORDER = 3,
  DELIVERED = 4
}

export enum ORDER_ACTION_TYPE {
  UPLOAD = "upload",
  CONFIRM = "confirm",
  DELETE = "delete",
  EXPORT = "export",
  EXPORT_SFTP = "exportSftp",
  UNCONFIRMED = "unconfirmed",
  VALIDATE = "validate"
}

export enum Views {
  CUSTOMER = "customer",
  SUPPLIER = "supplier"
}

export const events = {
  CHANGE_ORDER_TYPE: "change:order-type"
};
export const supplierOrdersHeaders = [
  { name: ORDER_TYPE.SALES_ORDER, value: ORDER_TYPE_VALUE.SALES_ORDER },
  { name: ORDER_TYPE.DELIVERED, value: ORDER_TYPE_VALUE.DELIVERED }
];

export const userOrdersHeaders = [
  { name: ORDER_TYPE.PLANNED, value: ORDER_TYPE_VALUE.PLANNED },
  { name: ORDER_TYPE.VALIDATED, value: ORDER_TYPE_VALUE.VALIDATED },
  {
    name: ORDER_TYPE.PURCHASE_ORDER,
    value: ORDER_TYPE_VALUE.PURCHASE_ORDER
  },
  { name: ORDER_TYPE.DELIVERED, value: ORDER_TYPE_VALUE.DELIVERED }
];
