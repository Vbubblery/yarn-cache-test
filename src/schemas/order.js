export default {
  type: "object",
  properties: {
    id: {
      type: "integer",
      minimum: 1
    },
    status_id: {
      type: "integer",
      minimum: 1,
      maximum: 4
    },
    supplier_id: {
      type: "integer",
      minimum: 1
    },
    date: {
      type: "string"
    },
    supplies: {
      type: "object"
    }
  },
  required: ["id", "status_id", "supplier_id", "date", "supplies"]
};

export const orderSupplySchema = {
  type: "object",
  properties: {
    id: {
      type: "integer",
      minimum: 1
    },
    product_site: {
      type: "integer",
      minimum: 1
    },
    quantity: {
      type: "integer"
    },
    order: {
      type: "integer",
      minimum: 1
    },
    name: {
      type: "string"
    }
  },
  required: ["id", "product_site", "quantity", "name"]
};
