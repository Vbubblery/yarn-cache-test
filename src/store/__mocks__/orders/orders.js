import ordersData from "./modules/ordersData";
import ordersDataOptions from "./modules/ordersDataOptions";

const namespaced = true;

const modules = {
  ordersData,
  ordersDataOptions
};

export default {
  namespaced,
  modules
};
