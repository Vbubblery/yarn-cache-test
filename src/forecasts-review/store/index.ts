import { computed, SetupContext, ComputedRef } from "@vue/composition-api";
import { PLANNING_DEMAND_VIEW } from "@/forecasts-review/typings";
import { Site } from "@/objects/site";

export const useBaseResources = (context: SetupContext) => {
  const store = context.parent?.$store || null;

  const getMe = computed<{ locale: string }>(
    () => store?.getters["common/getMe"]
  );
  const planningDemandView: ComputedRef<PLANNING_DEMAND_VIEW> = computed(
    () => store?.getters["common/getDefaultView"]
  );
  const currentSite = computed<Site>(
    () => store?.getters["common/getCurrentSite"]
  );
  const tablesData = computed(() => store?.getters["demand/getTablesData"]);
  const demandGetChartData = computed(
    () => store?.getters["demand/getChartData"]
  );
  const demandGetIsLoadingChart = computed(
    () => store?.getters["demand/getIsLoadingChart"]
  );
  const demandGetIsLoadingTables = computed(
    () => store?.getters["demand/getIsLoadingTables"]
  );
  const planningGetChartData = computed(
    () => store?.getters["planning/getChartData"]
  );
  const planningGetIsLoadingChart = computed(
    () => store?.getters["planning/getIsLoadingChart"]
  );
  const planningGetIsLoadingTables = computed(
    () => store?.getters["planning/getIsLoadingTables"]
  );
  const selectedTagIds = computed(
    () => store?.getters["common/getSelectedTags"]
  );
  const selectedCustomerId = computed(() =>
    currentSite.value.isSupplierView
      ? store?.getters["demand/getSelectedCustomerSiteId"]
      : null
  );
  const updateDefaultView = (view: PLANNING_DEMAND_VIEW) => {
    store?.commit("common/updateDefaultView", view);
  };
  const updateSelectedProducts = (items: object[]) => {
    store?.dispatch("common/updateSelectedProducts", items);
  };
  const updateSelectedTags = (items: object[]) => {
    store?.dispatch("common/updateSelectedTags", items);
  };
  const updateDataType = (itemType: string) => {
    store?.commit("common/updateDataType", itemType);
  };
  const resetDemandState = () => {
    store?.commit("demand/resetState");
  };
  const updateDemandPage = (pageNum: number) => {
    store?.commit("demand/updatePage", pageNum);
  };
  const updateDemandTableOnly = (onlyTable: boolean) => {
    store?.commit("demand/updateOnlyTable", onlyTable);
  };
  const fetchData = (
    selectedView: undefined | PLANNING_DEMAND_VIEW = undefined
  ) => {
    const currentView = selectedView ?? planningDemandView.value;
    store?.dispatch(
      currentView === PLANNING_DEMAND_VIEW.DEMAND
        ? "demand/fetchDemandData"
        : "planning/updatePlanningData"
    );
  };
  const updateSelectedCustomerSiteId = (customerSiteId: number | null) => {
    store?.commit("demand/updateSelectedCustomerSiteId", customerSiteId);
  };
  return {
    getMe,
    planningDemandView,
    updateDefaultView,
    currentSite,
    tablesData,
    demandGetChartData,
    demandGetIsLoadingChart,
    demandGetIsLoadingTables,
    planningGetChartData,
    planningGetIsLoadingChart,
    planningGetIsLoadingTables,
    selectedCustomerId,
    selectedTagIds,
    updateSelectedProducts,
    updateSelectedTags,
    updateDataType,
    resetDemandState,
    updateDemandPage,
    updateDemandTableOnly,
    fetchData,
    updateSelectedCustomerSiteId
  };
};
