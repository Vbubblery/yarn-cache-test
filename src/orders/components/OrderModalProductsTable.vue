<template>
  <div class="flex-1 flex flex-col overflow-hidden">
    <div class="grid pt-6 grid-cols-24 text-sm text-fl-dark-grey">
      <div
        class="col-start-2 p-2 pl-0"
        :class="currentSite.isSupplierView ? 'col-span-13' : 'col-span-8'"
      >
        {{ $t("orders.reference") }}
      </div>
      <div class="col-span-3 p-2 text-right">
        {{ $t("orders.quantity") }}
      </div>
      <div v-if="!currentSite.isSupplierView" class="col-span-3 p-2 text-right">
        {{ $t("orders.daysOfCoverage") }}
      </div>
      <div class="col-span-2 p-2 text-right">
        {{ $t("orders.lotSize") }}
      </div>
      <div class="col-span-2 p-2 text-right">
        {{ $t("orders.moq") }}
      </div>
      <div class="col-span-3 p-2 pr-8 text-right">
        {{ $t("orders.value") }}
      </div>
    </div>
    <div class="flex-1 overflow-auto pt-[1px] pb-1">
      <OrderModalProductsTableItem
        v-for="supply in order.supplies"
        :key="`${order.id}-${supply.id}`"
        :order="order"
        :supply="supply"
        :is-editing="isEditing"
        :view="view"
        :editable-supplies="editableSupplies"
        :editable-supply-quantity="editableSupplies[supply.productId]"
        v-on="$listeners"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { Site } from "@/objects/site";
import { Order } from "@/objects/order";
import { computed, defineComponent } from "@vue/composition-api";
import OrderModalProductsTableItem from "@/orders/components/OrderModalProductsTableItem.vue";

export default defineComponent({
  name: "OrderModalProductsTable",
  components: {
    OrderModalProductsTableItem
  },
  props: {
    order: {
      required: true,
      type: Object as () => Order
    },
    isEditing: {
      required: true,
      type: Boolean
    },
    editableSupplies: {
      required: true,
      type: Object as () => Record<string, number>
    },
    view: {
      required: true,
      type: String
    }
  },
  setup(_, context) {
    const store = context.root.$store;
    const currentSite = computed<Site>(
      () => store.getters["common/getCurrentSite"]
    );

    return {
      currentSite
    };
  }
});
</script>
