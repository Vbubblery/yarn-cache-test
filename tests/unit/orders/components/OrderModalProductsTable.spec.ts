import { renderComponent } from "tests/test-utils";
import OrderModalProductsTable from "@/orders/components/OrderModalProductsTable.vue";
import { Order } from "@/objects/order";
import { Views } from "@/orders/constants";

// eslint-disable-next-line jest/no-mocks-import
import { store } from "@/store/__mocks__/store";

/**
 * This is a "dumb" component. It has no responsibility besides
 * displaying in a table format the products of an order.
 * As such, a snapshot test is enough to raise a warning if there is a UI change
 */
describe("OrderModalProductsTable", () => {
  it("Renders the component", () => {
    const { html } = renderComponent(OrderModalProductsTable, {
      props: {
        order: new Order(),
        isEditing: false,
        editableSupplies: {},
        view: Views.CUSTOMER
      },
      store
    });
    expect(html()).toMatchSnapshot();
  });
});
