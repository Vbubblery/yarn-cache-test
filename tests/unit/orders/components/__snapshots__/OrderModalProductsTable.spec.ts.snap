// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderModalProductsTable Renders the component 1`] = `
<div class="flex-1 flex flex-col overflow-hidden">
  <div class="grid pt-6 grid-cols-24 text-sm text-fl-dark-grey">
    <div class="col-start-2 p-2 pl-0 col-span-8">
      orders.reference
    </div>
    <div class="col-span-3 p-2 text-right">
      orders.quantity
    </div>
    <div class="col-span-3 p-2 text-right">
      orders.daysOfCoverage
    </div>
    <div class="col-span-2 p-2 text-right">
      orders.lotSize
    </div>
    <div class="col-span-2 p-2 text-right">
      orders.moq
    </div>
    <div class="col-span-3 p-2 pr-8 text-right">
      orders.value
    </div>
  </div>
  <div class="flex-1 overflow-auto pt-[1px] pb-1"></div>
</div>
`;
